Common/Data/Consolidators/PsychSignalConsolidator.cs

```diff
@@ -0,0 +1,122 @@
+using QuantConnect.Data.Market;
+
+namespace QuantConnect.Data.Consolidators
+{
+    /// <summary>
+    /// Consolidates PsychSignal data to the time period specified
+    /// </summary>
+    public class PsychSignalConsolidator : PeriodCountConsolidatorBase<PsychSignalSentimentData, PsychSignalConsolidated>
+    {
+        /// <summary>
+        /// Initializes a new instance of the <see cref="PsychSignalConsolidator"/> class
+        /// </summary>
+        /// <param name="period">The minimum span of time before emitting <see cref="PsychSignalConsolidated"/></param>
+        public PsychSignalConsolidator(TimeSpan period)
+            : base(period)
+        {
+        }
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="PsychSignalConsolidator"/> class
+        /// </summary>
+        /// <param name="maxCount">The number of pieces to accept before emiting <see cref="PsychSignalConsolidated"/></param>
+        public PsychSignalConsolidator(int maxCount)
+            : base(maxCount)
+        {
+        }
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="PsychSignalConsolidator"/> class
+        /// </summary>
+        /// <param name="maxCount">The number of pieces to accept before emiting <see cref="PsychSignalConsolidated"/></param>
+        /// <param name="period">The minimum span of time before emitting <see cref="PsychSignalConsolidated"/></param>
+        public PsychSignalConsolidator(int maxCount, TimeSpan period)
+            : base(maxCount, period)
+        {
+        }
+
+        /// <summary>
+        /// Creates a consolidator to produce a new 'PsychSignalConsolidated' representing the last count pieces of data or the period, whichever comes first
+        /// </summary>
+        /// <param name="func">Func that defines the start time of a consolidated data</param>
+        public PsychSignalConsolidator(Func<DateTime, CalendarInfo> func)
+            : base(func)
+        {
+        }
+
+        /// <summary>
+        /// Aggregates the new 'data' into the 'workingBar'. The 'workingBar' will be
+        /// null following the event firing
+        /// </summary>
+        /// <param name="workingBar">The <see cref="PsychSignalConsolidated"/> instance we're building, null if the event was just fired and we're starting a new instance</param>
+        /// <param name="data">The new data</param>
+        protected override void AggregateBar(ref PsychSignalConsolidated workingBar, PsychSignalSentimentData data)
+        {
+            var bullIntensity = data.BullIntensity;
+            var bearIntensity = data.BearIntensity;
+            var bullMinusBear = data.BullMinusBear;
+            var bullScoredMessages = data.BullScoredMessages;
+            var bearScoredMessages = data.BearScoredMessages;
+            var bullBearMessageRatio = data.BullBearMessageRatio;
+            var totalScoredMessages = data.TotalScoredMessages;
```

Comment:
why define these here and then not use them when initializing `workingBar`?

I think this function will read better by simply in-lining all of these variables